pkg/cmd/adm/register_member.go

```diff
@@ -29,93 +38,177 @@ const (
+	memberClusterClient, err := ctx.NewClientFromRestConfig(memberClientConfig)
+	if err != nil {
 		return err
 	}
 
-	warningMessage := "The automatic approval was disabled!\n Configure the new member cluster in ToolchainConfig and apply the changes to the cluster."
+	// as the first thing, let's check that the member has not been yet registered with the cluster.
+	hostsInMember := &toolchainv1alpha1.ToolchainClusterList{}
+	if err = memberClusterClient.List(ctx, hostsInMember, runtimeclient.InNamespace(memberOperatorNamespace)); err != nil {
+		return err
+	}
+	if len(hostsInMember.Items) > 0 {
+		return fmt.Errorf("the member cluster (%s) is already registered with some host cluster in namespace %s", memberApiEndpoint, memberOperatorNamespace)
+	}
 
-	if err := restartHostOperator(ctx, hostClusterClient, hostClusterConfig); err != nil {
-		return fmt.Errorf("%w\nIn Additon, there is another warning you should be aware of:\n%s", err, warningMessage)
+	// add the host entry to the member cluster first. We assume that there is just 1 toolchain cluster entry in the member
+	// cluster (i.e. it just points back to the host), so there's no need to determine the number of entries with the same
+	// API endpoint.
+	hostToolchainClusterName, err := utils.GetToolchainClusterName(string(configuration.Host), hostApiEndpoint, 0)
+	if err != nil {
+		return err
+	}
+	hostToolchainClusterKey := runtimeclient.ObjectKey{
+		Name:      hostToolchainClusterName,
+		Namespace: memberOperatorNamespace,
+	}
+	if err := runAddClusterScript(ctx, newCommand, configuration.Host, hostKubeconfig, hostOperatorNamespace, memberKubeconfig, memberOperatorNamespace, 0, useLetsEncrypt); err != nil {
+		return err
 	}
 
-	ctx.Printlnf("!!!!!!!!!!!!!!!")
-	ctx.Printlnf("!!! WARNING !!!")
-	ctx.Printlnf("!!!!!!!!!!!!!!!")
-	ctx.Printlnf(warningMessage)
-	return nil
-}
+	if err := waitUntilToolchainClusterReady(ctx.CommandContext, memberClusterClient, hostToolchainClusterKey, waitForReadyTimeout); err != nil {
+		ctx.Println("The ToolchainCluster resource representing the host in the member cluster has not become ready.")
+		ctx.Printlnf("Please check the %s ToolchainCluster resource in the %s member cluster.", hostToolchainClusterKey, memberApiEndpoint)
+		return err
+	}
 
-func disableAutomaticApproval(hostClusterConfig configuration.ClusterConfig, cl runtimeclient.Client) error {
-	configs := &toolchainv1alpha1.ToolchainConfigList{}
-	if err := cl.List(context.TODO(), configs, runtimeclient.InNamespace(hostClusterConfig.OperatorNamespace)); err != nil {
+	// figure out the name that will be given to our new ToolchainCluster representing the member in the host cluster.
+	// This is the same name that the add-cluster.sh script will deduce and use.
+	ord, err := getNumberOfToolchainClustersWithHostname(ctx, hostClusterClient, memberApiEndpoint, hostOperatorNamespace)
+	if err != nil {
+		return err
+	}
```

Comment:
it's the same as above - it's totally ok to run the command multiple times for the same combination of host & member cluster. 
Let's not try to make the command too smart and let's just pass the multi-member suffix as a flag (in the same way as we do it in the script).